src/main/java/com/google/api/codegen/util/ProtoParser.java

```diff
@@ -166,18 +165,6 @@ public String getResourceReferenceName(
     String resourceName = getResourceReference(field);
     if (!Strings.isNullOrEmpty(resourceName)) {
 
-      TypeRef resourceType = field.getModel().getSymbolTable().lookupType(resourceName);
```

Comment:
This code block ends up being unnecessary, and a bug, because the `allResources` and `allResourceSets` objects already contain the message-defined Resource[Set]s.